beacon-chain/state/v2/getters_validator.go

```diff
@@ -182,12 +161,6 @@ func (b *BeaconState) NumValidators() int {
 // ReadFromEveryValidator reads values from every validator and applies it to the provided function.
 // Warning: This method is potentially unsafe, as it exposes the actual validator registry.
 func (b *BeaconState) ReadFromEveryValidator(f func(idx int, val state.ReadOnlyValidator) error) error {
-	if !b.hasInnerState() {
-		return ErrNilInnerState
-	}
-	if b.validators == nil {
-		return errors.New("nil validators in state")
-	}
```

Comment:
Unsure about this one. In `develop` we have https://github.com/prysmaticlabs/prysm/blob/00c3a7dcaf2f66b44b292289dff9bf00a66ca60c/beacon-chain/db/kv/state.go#L430
and `InitializeFromProtoUnsafe` skips proto cloning. An empty array field value returns in `nil` after the cloning. 

This test failed the nil check: https://github.com/prysmaticlabs/prysm/blob/00c3a7dcaf2f66b44b292289dff9bf00a66ca60c/beacon-chain/rpc/prysm/v1alpha1/beacon/assignments_test.go#L47 
This is because it had zero validators in the state. The question is what to do now. Add a validator to this test? Assign an empty validators array when we get nil after proto cloning (note that this will assign the empty array also in case where we had nil originally)?

Reply:
OK, now another test fails because it expects the nil validators error...